src/ast/mod.rs

```diff
@@ -8543,27 +8617,13 @@ impl Display for CreateViewParams {
     }
 }
 
-/// Engine of DB. Some warehouse has parameters of engine, e.g. [clickhouse]
-///
-/// [clickhouse]: https://clickhouse.com/docs/en/engines/table-engines
 #[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
-pub struct TableEngine {
-    pub name: String,
-    pub parameters: Option<Vec<Ident>>,
-}
-
-impl Display for TableEngine {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", self.name)?;
-
-        if let Some(parameters) = self.parameters.as_ref() {
-            write!(f, "({})", display_comma_separated(parameters))?;
-        }
-
-        Ok(())
-    }
+pub struct NamedParenthesizedList {
+    pub key: Ident,
+    pub value: Option<Ident>,
+    pub parameters: Vec<Ident>,
```

Comment:
hmm not sure re the naming of the properties, they don't hint anything being named as the `NamedParenthesizeList` suggest and the parameters property is essentially a list of values as well so that the current struct is declaring a `value` and `values` which feels odd. maybe replacing `value` with `name`, and `parameters` with `values` would make more sense? Since currently if `value` is None, then the `Vec` isn't representing parameters for that option, rather a plain list of values